UI/window-basic-main-outputs.cpp

```diff
@@ -1977,6 +1983,48 @@ bool AdvancedOutput::StartReplayBuffer()
 	return true;
 }
 
+void AdvancedOutput::SaveReplayBuffer()
+{
+	if (!ffmpegOutput || ffmpegRecording) {
+		const char *path = config_get_string(
+			main->Config(), "AdvOut",
+			ffmpegRecording ? "FFFilePath" : "RecFilePath");
+		const char *recFormat = config_get_string(
+			main->Config(), "AdvOut",
+			ffmpegRecording ? "FFExtension" : "RecFormat");
+		const char *filenameFormat = config_get_string(
+			main->Config(), "Output", "FilenameFormatting");
+		bool overwriteIfExists = config_get_bool(
+			main->Config(), "Output", "OverwriteIfExists");
+		bool noSpace = config_get_bool(
+			main->Config(), "AdvOut",
+			ffmpegRecording ? "FFFileNameWithoutSpace"
+					: "RecFileNameWithoutSpace");
+		const char *rbPrefix = config_get_string(
+			main->Config(), "SimpleOutput", "RecRBPrefix");
+		const char *rbSuffix = config_get_string(
+			main->Config(), "SimpleOutput", "RecRBSuffix");
+
+		string f = GetFormatString(filenameFormat, rbPrefix, rbSuffix);
+		string strPath = GetOutputFilename(
+			path, recFormat, noSpace, overwriteIfExists, f.c_str());
+		QFileInfo fileInfo(QString::fromStdString(strPath));
+		QString fileName(fileInfo.completeBaseName());
+
+		if (!fileName.isEmpty()) {
```

Comment:
Same here.  You can do this instead:

```
	if (fileName.isempty())
		return;
```

And then you can work with the obs_data after that.